documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -156,15 +156,15 @@ All actor methods are called in the order that the state changes happened in the
 
 **INFORMATIVE TEXT:** Some backends, such as OpenXR, may only operate on a state query basis, and obviously we only query the state when `Update` is called. This is the reason why it's only a "where possible" requirement, because if we can't actually get the events in the order they happened from the underlying backend (like we can for GLFW, for example) then it's impossible for us to convey this up to the developer. However, regardless of backend-specifics, a `State` update must always be paired with an actor method call.
 
-The `IEventReceiver` passed into `Update` may implement multiple other actor interfaces (as defined below), and if the actor implements an extra interface (such as `IMouseEventReceiver` defined below) that would allow the backend to forward more events to the actor, the backend must do so via type checking. That is, if `actor` is an instance of `IMouseEventReceiver`, any mouse events are delivered to that actor. But if `actor` does not implement `IMouseEventReceiver`, no mouse events will be delivered.
+The `IInputHandler` passed into `Update` may implement multiple other actor interfaces (as defined below), and if the actor implements an extra interface (such as `IMouseInputHandler` defined below) that would allow the backend to forward more events to the actor, the backend must do so via type checking. That is, if `actor` is an instance of `IMouseInputHandler`, any mouse events are delivered to that actor. But if `actor` does not implement `IMouseInputHandler`, no mouse events will be delivered.
```

Comment:
We should probably add that any events that could be raised by an interface that is not implemented by the handler should be discarded on `Update()`, to make sure we aren't leaving open the possibility of a managed memory leak

Reply:
```suggestion
The `IInputHandler` passed into `Update` may implement multiple other actor interfaces (as defined below), and if the actor implements an extra interface (such as `IMouseInputHandler` defined below) that would allow the backend to forward more events to the actor, the backend must do so via type checking. That is, if `actor` is an instance of `IMouseInputHandler`, any mouse events are delivered to that actor. But if `actor` does not implement `IMouseInputHandler`, no mouse events will be delivered. All events, including those that were not delivered due to the actor not implementing a necessary interface, must be discarded at the end of the `Update` call.
```